flink-connectors/flink-connector-kafka-0.9/src/test/java/org/apache/flink/streaming/connectors/kafka/Kafka09FetcherTest.java

```diff
@@ -422,6 +429,99 @@ public void run() {
 		assertFalse("fetcher threads did not properly finish", sourceContext.isStillBlocking());
 	}
 
+	@Test
+	public void testRichDeserializationSchema() throws Exception {
```

Comment:
I think we should enhance this test to test the behaviour with multiple `collect`s per record also.